pkg/api/types.go

```diff
@@ -0,0 +1,151 @@
+package api
+
+// TypeMeta describes an individual API model object
+type TypeMeta struct {
+	// APIVersion is on every object
+	APIVersion string `json:"apiVersion"`
+}
+
+// ResourcePurchasePlan defines resource plan as required by ARM
+// for billing purposes.
+type ResourcePurchasePlan struct {
+	Name          string `json:"name"`
+	Product       string `json:"product"`
+	PromotionCode string `json:"promotionCode"`
+	Publisher     string `json:"publisher"`
+}
+
+// ContainerService complies with the ARM model of
+// resource definition in a JSON template.
+type ContainerService struct {
+	ID       string                `json:"id"`
+	Location string                `json:"location"`
+	Name     string                `json:"name"`
+	Plan     *ResourcePurchasePlan `json:"plan,omitempty"`
+	Tags     map[string]string     `json:"tags"`
+	Type     string                `json:"type"`
+
+	Properties *Properties `json:"properties,omitempty"`
+}
+
+// Properties represents the ACS cluster definition
+type Properties struct {
+	ProvisioningState       ProvisioningState        `json:"provisioningState,omitempty"`
+	OrchestratorProfile     *OrchestratorProfile     `json:"orchestratorProfile,omitempty"`
+	MasterProfile           *MasterProfile           `json:"masterProfile,omitempty"`
+	AgentPoolProfiles       []*AgentPoolProfile      `json:"agentPoolProfiles,omitempty"`
+	ServicePrincipalProfile *ServicePrincipalProfile `json:"servicePrincipalProfile,omitempty"`
+	AzProfile               *AzProfile               `json:"azProfile,omitempty"`
```

Comment:
@pweil- `AzProfile` doesn't need to be in `Properties` as the RP already has this metadata and can be passed separately to the OSA plugin.

Reply:
See this conversation: https://github.com/openshift/openshift-azure/pull/143#discussion_r207224271  

I think if we can settle on how it is enriched (either by throwing it on the converted profile or as part of interface method args) we will know if it can be removed.  Which do you prefer?

cc @kargakis